packages/headless/src/features/result-preview/result-preview-analytics-actions.test.ts

```diff
@@ -0,0 +1,78 @@
+import {createRelay} from '@coveo/relay';
+import {CoveoSearchPageClient} from 'coveo.analytics';
+import {
+  SearchEngine,
+  buildSearchEngine,
+  getSampleSearchEngineConfiguration,
+} from '../../app/search-engine/search-engine';
+import {buildMockNonEmptyResult} from '../../test/mock-result';
+import {logDocumentQuickview} from './result-preview-analytics-actions';
+
+jest.mock('@coveo/relay');
+jest.mock('coveo.analytics');
+
+async function wait() {
+  return new Promise((resolve) => setTimeout(resolve, 0));
+}
+
+describe('#logRecommendationOpen', () => {
+  const testResult = buildMockNonEmptyResult();
+  let engine: SearchEngine;
+  const makeDocumentQuickview = jest.fn();
+  const emit = jest.fn();
+
+  beforeEach(() => {
+    jest.mocked(CoveoSearchPageClient).mockReturnValue({
+      makeDocumentQuickview,
+    } as unknown as CoveoSearchPageClient);
+    jest.mocked(createRelay).mockReturnValue({
+      emit,
+      getMeta: jest.fn(),
+      on: jest.fn(),
+      off: jest.fn(),
+      updateConfig: jest.fn(),
+      clearStorage: jest.fn(),
+      version: 'foo',
+    });
+  });
+
+  describe('when analyticsMode is `legacy`', () => {
+    beforeEach(() => {
+      engine = buildSearchEngine({
+        configuration: {
+          ...getSampleSearchEngineConfiguration(),
+        },
+      });
+    });
+
+    it('should call coveo.analytics.makeRecommendationOpen properly', async () => {
+      engine.dispatch(logDocumentQuickview(testResult));
+      await wait();
```

Comment:
Mmm.. why is this necessary?

Reply:
There are 'unbound' promises (i.e. out-of-reach promises) that we need to wait for.
This is a trick that allows us to flush the microtask queue to an extent.

Reply:
In the meantime, might be worth adding a little comment (maybe where we define the wait() function) to explain this cause it's not super obvious 😅